openmetadata-service/src/main/java/org/openmetadata/service/util/OpenMetadataOperations.java

```diff
@@ -260,49 +263,86 @@ public Integer reIndex(
-                appRunRecord.getSuccessContext().toString(),
-                appRunRecord.getFailureContext().toString()));
-        printToAsciiTable(columns, rows, "No Search Indexing Application found");
-      }
-      return 0;
+      return waitAndReturnReindexingAppStatus(searchIndexApp);
     } catch (Exception e) {
       LOG.error("Failed to reindex due to ", e);
       return 1;
     }
   }
 
+  private App getSearchIndexingApp(String appName, int batchSize, boolean recreateIndexes) {
+    try {
+      AppRepository appRepository = (AppRepository) Entity.getEntityRepository(Entity.APPLICATION);
+      App searchApp =
+          appRepository.getByName(null, "SearchIndexingApplication", appRepository.getFields("id"));
+      return searchApp;
+    } catch (EntityNotFoundException ex) {
+      return new App()
+          .withId(UUID.randomUUID())
+          .withName(appName)
+          .withFullyQualifiedName(appName)
+          .withClassName("org.openmetadata.service.apps.bundles.searchIndex.SearchIndexApp")
+          .withAppSchedule(new AppSchedule().withScheduleTimeline(ScheduleTimeline.DAILY))
+          .withAppConfiguration(
+              new EventPublisherJob()
+                  .withEntities(new HashSet<>(List.of("all")))
+                  .withRecreateIndex(recreateIndexes)
+                  .withBatchSize(batchSize)
+                  .withSearchIndexMappingLanguage(
+                      config.getElasticSearchConfiguration().getSearchIndexMappingLanguage()))
+          .withRuntime(new ScheduledExecutionContext().withEnabled(true));
+    }
+  }
+
+  @SneakyThrows
+  private int waitAndReturnReindexingAppStatus(App searchIndexApp) {
+    AppRunRecord appRunRecord;
+    do {
+      try {
+        AppRepository appRepository =
+            (AppRepository) Entity.getEntityRepository(Entity.APPLICATION);
+        appRunRecord = appRepository.getLatestAppRuns(searchIndexApp.getId());
+        if (appRunRecord != null) {
+          List<String> columns =
+              new ArrayList<>(
+                  List.of("status", "startTime", "endTime", "executionTime", "success", "failure"));
+          List<List<String>> rows = new ArrayList<>();
+          rows.add(
+              Arrays.asList(
+                  appRunRecord.getStatus().value(),
+                  appRunRecord.getStartTime().toString(),
+                  appRunRecord.getEndTime().toString(),
+                  appRunRecord.getExecutionTime().toString(),
+                  nullOrEmpty(appRunRecord.getSuccessContext())
+                      ? "Unavailable"
+                      : JsonUtils.pojoToJson(appRunRecord.getSuccessContext()),
+                  nullOrEmpty(appRunRecord.getFailureContext())
+                      ? "Unavailable"
+                      : JsonUtils.pojoToJson(appRunRecord.getFailureContext())));
+          printToAsciiTable(columns, rows, "No Search Indexing Application found");
```

Comment:
Lets change the error message to "Failed to run Search Reindexing"

Reply:
Done @harshach